src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/allelespecific/AS_RankSumTest.java

```diff
@@ -44,44 +58,55 @@
             return Collections.emptyMap();
         }
 
-        final Map<String, Object> annotations = new HashMap<>();
-        final AlleleSpecificAnnotationData<CompressedDataList<Integer>> myData = initializeNewAnnotationData(vc.getAlleles());
-        calculateRawData(vc, likelihoods, myData);
-        final String annotationString = makeRawAnnotationString(vc.getAlleles(), myData.getAttributeMap());
+        final AlleleSpecificAnnotationData<CompressedDataList<Integer>> myRawData = initializeNewRawAnnotationData(vc.getAlleles());
+        calculateRawData(vc, likelihoods, myRawData);
+        Map<Allele, List<Double>> myRankSumStats = calculateRankSum(myRawData.getAttributeMap(), myRawData.getRefAllele());
+        final String annotationString = makeRawAnnotationString(vc.getAlleles(),myRankSumStats);
         if (annotationString == null){
             return Collections.emptyMap();
         }
-        annotations.put(getRawKeyName(), annotationString);
-        return annotations;
+        return Collections.singletonMap(getRawKeyName(), annotationString);
     }
 
-    private AlleleSpecificAnnotationData<CompressedDataList<Integer>> initializeNewAnnotationData(final List<Allele> vcAlleles) {
-        final Map<Allele, CompressedDataList<Integer>> perAlleleValues = new HashMap<>();
-        for (final Allele a : vcAlleles) {
-            perAlleleValues.put(a, new CompressedDataList<>());
+    //TODO figure out if this needs to be a double to behave correctly
```

Comment:
I had to iterate on this one a few times and there appear to be some loose ends.  I think CompressedDataList<Integer> should be replaced with Histogram.  Did you port https://github.com/broadinstitute/gsa-unstable/pull/1610 and https://github.com/broadinstitute/gsa-unstable/pull/1541 already?

Reply:
-It looks like I have ported broadinstitute/gsa-unstable#1541 already, but it appears I have missed some of the changes in broadinstitute/gsa-unstable#1610. I will port and test that now.